crates/rari-doc/src/pages/page.rs

```diff
@@ -110,17 +110,45 @@ impl Page {
         } = url_meta_from(url)?;
         let locale = locale.unwrap_or(locale_from_url);
         match page_category {
-            PageCategory::SPA => SPA::from_slug(slug, locale)
-                .ok_or(DocError::PageNotFound(url.to_string(), PageCategory::SPA)),
+            PageCategory::SPA => {
+                if locale != Locale::EnUs && slug.starts_with("blog") {
+                    // Blog is en-US only.
+                    Err(DocError::PageNotFound(url.to_string(), PageCategory::SPA))
+                } else {
+                    SPA::from_slug(slug, locale)
+                        .ok_or(DocError::PageNotFound(url.to_string(), PageCategory::SPA))
+                }
+            }
             PageCategory::Doc => Doc::page_from_slug_path(&folder_path, locale, fallback)
                 .map_err(|_| DocError::PageNotFound(url.to_string(), PageCategory::Doc)),
-            PageCategory::BlogPost => BlogPost::page_from_url(url).ok_or(DocError::PageNotFound(
-                url.to_string(),
-                PageCategory::BlogPost,
-            )),
-            PageCategory::Curriculum => Curriculum::page_from_url(url).ok_or(
-                DocError::PageNotFound(url.to_string(), PageCategory::Curriculum),
-            ),
+            PageCategory::BlogPost => {
+                if locale != Locale::EnUs {
+                    // Blog is en-US only.
+                    Err(DocError::PageNotFound(
+                        url.to_string(),
+                        PageCategory::BlogPost,
+                    ))
+                } else {
+                    BlogPost::page_from_url(url).ok_or(DocError::PageNotFound(
+                        url.to_string(),
+                        PageCategory::BlogPost,
+                    ))
+                }
+            }
```

Comment:
A bit easier to read.
```suggestion
            PageCategory::BlogPost if locale != Locale::EnUs => {
                // Blog is en-US only.
                Err(DocError::PageNotFound(
                    url.to_string(),
                    PageCategory::BlogPost,
                ))
            PageCategory::BlogPost => {
                BlogPost::page_from_url(url).ok_or(DocError::PageNotFound(
                    url.to_string(),
                    PageCategory::BlogPost,
                ))
            }
```
Same for curriculum.

Reply:
Thanks, applied in 95f94abcc1c30afabc227d95e185354455aaad98.